### PR TITLE
Change CopyOnWriteArrayList to CopyOnWriteArraySet in the sdk to hold listeners.

### PR DESCRIPTION
**Pull request by pengdev**
### Summary of changes

This PR changes the `CopyOnWriteArrayList` to `CopyOnWriteArraySet` in the sdk to hold listeners. And avoid similar issues as in https://github.com/mapbox/mapbox-maps-android/issues/301 that duplicated subscription would be invoked multiple times.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `./gradlew apiDump` to update generated api files, if there's public API changes, otherwise the `verify-kotlin-binary-compatibility` CI will fail.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.
 - [x] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 1]
```diff
@@ -54,16 +54,16 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase {
   private lateinit var mapPluginProviderDelegate: MapPluginProviderDelegate
   private lateinit var cameraAnimationsPlugin: CameraAnimationsPlugin
 
-  private val protectedCameraAnimatorOwnerList = CopyOnWriteArrayList<String>()
+  private val protectedCameraAnimatorOwnerList = CopyOnWriteArraySet<String>()
```
Comment (yunikkk) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:57]:
```suggestion
  private val protectedCameraAnimatorOwners = CopyOnWriteArraySet<String>()
```

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 2]
```diff
@@ -54,16 +54,16 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase {
...
-  private val onRotateListenerList = CopyOnWriteArrayList<OnRotateListener>()
-  private val onScaleListenerList = CopyOnWriteArrayList<OnScaleListener>()
-  private val onShoveListenerList = CopyOnWriteArrayList<OnShoveListener>()
+  private val onMapClickListenerList = CopyOnWriteArraySet<OnMapClickListener>()
+  private val onMapLongClickListenerList = CopyOnWriteArraySet<OnMapLongClickListener>()
+  private val onFlingListenerList = CopyOnWriteArraySet<OnFlingListener>()
+  private val onMoveListenerList = CopyOnWriteArraySet<OnMoveListener>()
+  private val onRotateListenerList = CopyOnWriteArraySet<OnRotateListener>()
+  private val onScaleListenerList = CopyOnWriteArraySet<OnScaleListener>()
+  private val onShoveListenerList = CopyOnWriteArraySet<OnShoveListener>()
```
Comment (yunikkk) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:66]:
```suggestion
  private val onShoveListeners = CopyOnWriteArraySet<OnShoveListener>()
```

Comment (yunikkk) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:66]: And the rest

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 3]
```diff
@@ -1382,29 +1382,29 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase {
...
   internal fun handleSingleTapUpEvent(): Boolean {
-    cameraAnimationsPlugin.cancelAllAnimators(protectedCameraAnimatorOwnerList)
+    cameraAnimationsPlugin.cancelAllAnimators(protectedCameraAnimatorOwners)
     return true
   }
 
   private fun notifyOnFlingListeners() {
-    if (!onFlingListenerList.isEmpty()) {
-      for (listener in onFlingListenerList) {
+    if (!onFlingListeners.isEmpty()) {
```
Comment (alexander-kulikovskii) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1390]: do really need this check and all below?

Comment (pengdev) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt]: Removed these isEmpty checks

**Review by alexander-kulikovskii — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
+# 10.4.0-rc.1
```
Comment (yunikkk) [inline CHANGELOG.md:5]: Shouldn't it be `main` since we don't know the next release yet?

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
+# main
```
Comment (ank27) [inline CHANGELOG.md:5]: nit : we can add it as `10.4.0-rc.1`. not required though. we can change that later :D

**Review by ank27 — APPROVED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (pengdev) [reply CHANGELOG.md:5]: It was just changed from `10.4.0-rc.1` to `main`. due to the comment in https://github.com/mapbox/mapbox-maps-android/pull/1183#discussion_r814007886

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,10 @@
...
+## Bug fixes 🐞
+* Change CopyOnWriteArrayList to CopyOnWriteArraySet in the sdk to hold listeners. ([1183](https://github.com/mapbox/mapbox-maps-android/pull/1183))
```
Comment (kiryldz) [inline CHANGELOG.md:7]: perhaps add more color that duplicates will be eliminated? however not sure here

Comment (kiryldz) [inline CHANGELOG.md:7]: perhaps add more color that duplicates will be eliminated? however not sure here

**Review by kiryldz — APPROVED** [review]
LGTM with a nit, nice catch!

**Pull request merged**